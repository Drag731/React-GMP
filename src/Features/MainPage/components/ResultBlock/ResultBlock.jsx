import React from 'react';

// import './ResultBlock.scss'

const ResultBlock = ({handleReleaseDateSort, handleRatingSort, total, sortBy}) => (
    <div className="main-page__result b-result">
        <div className="b-result__text-wrap">
            <span className="b-result__text">
                {total} movies found
            </span>
        </div>
        <div className="b-result__sort-wrap">
            <span className="b-result__sort-text">Sort by:</span>
            <span
                onClick={handleReleaseDateSort}
                className="b-result__sort-item"
                style={
                    {
                        color: sortBy === 'release_date' ? 'red' : '#000',
                        pointerEvents: sortBy === 'release_date' ? 'none' : 'auto'
                    }
                }
            >
                release date
            </span>
            <span
                onClick={handleRatingSort}
                className="b-result__sort-item"
                style={
                    {
                        color: sortBy === 'vote_average' ? 'red' : '#000',
                        pointerEvents: sortBy === 'vote_average' ? 'none' : 'auto'
                    }
                }
            >
                rating
            </span>
        </div>
    </div>
);

export default ResultBlock;
