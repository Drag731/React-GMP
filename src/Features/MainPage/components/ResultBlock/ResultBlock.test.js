import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ResultBlock from './ResultBlock.jsx'

configure({ adapter: new Adapter() });

describe('<ResultBlock  />', () => {
    it('Renders ResultBlock component', () => {
        const handleReleaseDateSort = jest.fn();
        const handleRatingSort = jest.fn();
        const total = 1;
        const tree = shallow(
            <ResultBlock
                handleReleaseDateSort={handleReleaseDateSort}
                handleRatingSort={handleRatingSort}
                total={total}
            />
        );
        expect(tree.find('.main-page__result')).toHaveLength(1);
    });
});