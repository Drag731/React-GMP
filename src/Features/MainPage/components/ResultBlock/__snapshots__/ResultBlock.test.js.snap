// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultBlock  /> Snapshot one 1`] = `
<div
  className="main-page__result b-result"
>
  <div
    className="b-result__text-wrap"
  >
    <span
      className="b-result__text"
    >
      1
       movies found
    </span>
  </div>
  <div
    className="b-result__sort-wrap"
  >
    <span
      className="b-result__sort-text"
    >
      Sort by:
    </span>
    <span
      className="b-result__sort-item"
      onClick={[MockFunction]}
    >
      release date
    </span>
    <span
      className="b-result__sort-item"
      onClick={[MockFunction]}
    >
      rating
    </span>
  </div>
</div>
`;
